log.h

```diff
@@ -41,7 +41,7 @@ int log_internal(int level, int error,
     ({                                                                  \
          int _level = (level), _e = (error);                            \
          (log_get_max_level() >= LOG_PRI(_level))                       \
-            ? log_internal(_level, _e, __FILE__, __LINE__, __VA_ARGS__) \
+            ? log_internal(_level, _e, __BASE_FILE__, __LINE__, __VA_ARGS__) \
```

Comment:
From gcc doc:

> This macro expands to the name of the main input file, in the form of a C string constant. This is the source file that was specified on the command line of the preprocessor or C compiler. 

So, AFAIU, this could still show `foo/bar/baz.cpp` if that's what was passed from build env to compiler.
You said that it's working - so it could be OK - but intended results may not be obtained in all build environments.
Note: clang doc says the same.